lib/avro/builder/types/type.rb

```diff
@@ -30,6 +30,27 @@ def configure_options(_options = {})
         def self.union_with_null(serialized)
           [:null, serialized]
         end
+
+        # Subclasses should override this method to check for the presence
+        # of required DSL attributes.
+        def validate!
+        end
+
+        private
+
+        def required_attribute_error!(attribute_name)
+          raise RequiredAttributeError.new(type: type_name,
+                                           attribute: attribute_name,
+                                           field: field && field.name,
+                                           name: @name)
+        end
+
+        def missing!(attribute_name)
```

Comment:
Would it make sense to call this something like `validate_required_attribute!`? It wasn't obvious to me based on the name whether this would do validation or just raise an exception.